### PR TITLE
EWPP-3721: CircaBC integration.

### PR DESCRIPTION
**Pull request by upchuk**

### CODE REVIEW

File: modules/oe_media_circabc/src/DocumentMediaFormHandler.php [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+   *
+   * @param array $element
+   *   The element.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state.
+   * @param array $complete_form
+   *   The form.
+   */
+  public static function validateCircaBcUrl(&$element, FormStateInterface $form_state, &$complete_form) {
+    $url = NestedArray::getValue($form_state->getValues(), $element['#parents'], $input_exists);
```
[the remark this answers is not in the document]
Comment (upchuk) [reply modules/oe_media_circabc/src/DocumentMediaFormHandler.php:170]: Same here.

File: modules/oe_media_circabc/src/Plugin/Field/FieldType/CircaBcReferenceItem.php [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  }
+
+  /**
+   * Return the URL to the actual remote file.
+   *
+   * @return \Drupal\Core\Url
+   *   The URL object.
+   */
+  public function getFileUrl(): Url {
+    return Url::fromUri(Settings::get('circabc')['url'] . '/d/d/workspace/SpacesStore/' . $this->get('uuid')->getValue() . '/file.bin');
```
Comment (sergepavle) [inline modules/oe_media_circabc/src/Plugin/Field/FieldType/CircaBcReferenceItem.php:41]: Does it make sense to keep in constant pattern of URL (with placeholders)?

File: modules/oe_media_circabc/src/Plugin/Field/FieldFormatter/CircaBcFileLink.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+ */
+class CircaBcFileLink extends FormatterBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    $element = [];
+    foreach ($items as $delta => $item) {
+      $uuid = $item->uuid;
```
Comment (sergepavle) [inline modules/oe_media_circabc/src/Plugin/Field/FieldFormatter/CircaBcFileLink.php:29]: We don't use that variable here.
```suggestion
```

File: modules/oe_media_circabc/src/Plugin/Field/FieldFormatter/CircaBcFileLink.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+      $new_element = [
+        '#theme' => 'file_link_formatter',
+        '#link' => [
+          '#type' => 'link',
+          '#title' => $item->filename,
+          '#url' => $item->getFileUrl(),
+        ],
+        '#size' => format_size($item->size),
```
Comment (sergepavle) [inline modules/oe_media_circabc/src/Plugin/Field/FieldFormatter/CircaBcFileLink.php:38]: This function is deprecated in 10.2 drupal core.

File: config/install/core.entity_form_display.media.document.default.yml [hunk 5]
```diff
@@ -15,7 +15,7 @@ bundle: document
 mode: default
 content:
   oe_media_file_type:
-    weight: 1
+    weight: 0
```
[the remark this answers is not in the document]
Comment (upchuk) [reply config/install/core.entity_form_display.media.document.default.yml:18]: I double checked and actually it's a good enough change to be in the main module as well. As a creation flow, first you select what file type you want and then come the other fields. This works well also if you dont have circa BC enabled. And we don't touch the update path in this module, only new installs.

File: modules/oe_media_circabc/src/Plugin/Field/FieldType/CircaBcReferenceItem.php (hunk 2 above)
Comment (upchuk) [reply modules/oe_media_circabc/src/Plugin/Field/FieldType/CircaBcReferenceItem.php:41]: Neah, it's used only here.

File: modules/oe_media_circabc/src/Plugin/Field/FieldFormatter/CircaBcFileLink.php (hunk 4 above)
Comment (upchuk) [reply modules/oe_media_circabc/src/Plugin/Field/FieldFormatter/CircaBcFileLink.php:38]: Yeah, ok, we don't have yet `ByteSizeMarkup`. It's removed in Drupal 11.

**Review by sergepavle — APPROVED** [review]

**Pull request merged**